### PR TITLE
PYTHON-4417 Fix env var in bump script

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: bump-and-tag.sh [hunk 1]
```diff
@@ -9,16 +9,16 @@ echo "Running bump, tag, and bump..."
 set -x
 
 # Bump
-perl -pi -e "s/version = \"$CURRENT_VERSION\"/version = \"$NEXT_VERSION\"/" pyproject.toml
+perl -pi -e "s/version = \"$CURRENT_VERSION\"/version = \"$NEW_VERSION\"/" pyproject.toml
 git add .
-git commit -a -m "BUMP ${NEXT_VERSION}" -s --gpg-sign=${GPG_KEY_ID}
+git commit -a -m "BUMP $$NEW_VERSION}" -s --gpg-sign=${GPG_KEY_ID}
```
Comment (NoahStapp) [inline bump-and-tag.sh:14]: Missing an opening `{`?

**Review by NoahStapp — CHANGES_REQUESTED** [review]

File: bump-and-tag.sh (hunk 1 above)
Comment (blink1073) [reply bump-and-tag.sh:14]: Fixed

Comment (NoahStapp) [reply bump-and-tag.sh:14]: Still missing on line 14 and 17.

Comment (blink1073) [reply bump-and-tag.sh]: Fixed

File: bump-and-tag.sh [hunk 2]
```diff
@@ -9,16 +9,16 @@ echo "Running bump, tag, and bump..."
...
-perl -pi -e "s/version = \"$CURRENT_VERSION\"/version = \"$NEXT_VERSION\"/" pyproject.toml
+perl -pi -e "s/version = \"${CURRENT_VERSION}\"/version = \"${NEW_VERSION}\"/" pyproject.toml
 git add .
-git commit -a -m "BUMP ${NEXT_VERSION}" -s --gpg-sign=${GPG_KEY_ID}
+git commit -a -m "BUMP ${NEW_VERSION}" -s --gpg-sign=${GPG_KEY_ID}
 
 # Tag
-git tag -a "${NEXT_VERSION}" -m "BUMP ${NEXT_VERSION}" -s --local-user=${GPG_KEY_ID}
-git show --no-patch "${NEXT_VERSION}"
+git tag -a "$$NEW_VERSION}" -m "BUMP ${NEW_VERSION}" -s --local-user=${GPG_KEY_ID}
```
Comment (NoahStapp) [inline bump-and-tag.sh:17]:
```suggestion
git tag -a "${NEW_VERSION}" -m "BUMP ${NEW_VERSION}" -s --local-user=${GPG_KEY_ID}
```

**Review by NoahStapp — CHANGES_REQUESTED** [review]

**Review by NoahStapp — APPROVED** [review]

**Pull request merged**